api/metric/common.go

```diff
@@ -14,17 +14,18 @@
 
 package metric
 
-import (
-	"go.opentelemetry.io/api/registry"
+import "sync/atomic"
+
+var (
+	instrumentID uint64
 )
 
-func registerMetric(name string, mtype MetricType, opts []Option, metric *Handle) {
-	var varOpts []registry.Option
+func registerInstrument(name string, kind Kind, opts []Option, inst *Instrument) {
```

Comment:
Should the function check bogus set ups like `NonNegative` for `GaugeKind` and warn/panic in such cases? That way it would be easier to spot when we did something wrong.